.github/workflows/check.yaml

```diff
@@ -18,6 +18,10 @@ jobs:
       - uses: actions/setup-node@v2
       - run: npm install -g @commitlint/cli @commitlint/config-conventional
       - name: "Check commits"
+        # We don't care about the commits because we squash merge, but if
+        # there's only one the default merge commit message is the single commit
+        # message
+        if: github.event.pull_request.commits == 1
```

Comment:
I can't find the docs for the `commits` element in the `pull_request` event docs, so not clear to me what this check is doing. If it means that there is only 1 commit in the PR, that doesn't seem right. We want to check the commit subject of the first commit IIUC.

https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request

Reply:
[It's the number of commits (GH doesn't publish any kind of schema for their REST API)](https://docs.github.com/en/rest/reference/pulls#get-a-pull-request)

> We want to check the commit subject of the first commit IIUC.

Why? As far as I can tell, when we click squash and:
* the PR has 1 commit, the default squash commit message is the commit message
* Otherwise, the default squash commit message is the PR title

Reply:
If it is the number of commits, then is the check omitted altogether when there is more that one commit? That doesn't seem like what we want.

WRT squashed commits, when there are multiple commits the person merging ought to edit the commit message to make sense in the context of the PR being a single commit. Since there's no additional check, they just have to get it right I guess.

Reply:
> If it is the number of commits, then is the check omitted altogether when there is more that one commit? That doesn't seem like what we want.

Only the _commit_ check is omitted if we have >1 commit, not the PR title check.

The goal of this entire check was to prevent the GitHub-generated commit title, of the _merged commit_, (i.e. the first line of the commit message) from not passing the check. With this PR, that's still enforced because of the behavior I mentioned above.

> WRT squashed commits, when there are multiple commits the person merging ought to edit the commit message to make sense in the context of the PR being a single commit. Since there's no additional check, they just have to get it right I guess.

I don't think it's ever necessary to edit the _title_ of the commit because it will match the PR title (or the single commit message like I mentioned).

But I'm going to close this PR, because I made it only to _loosen_ the check for the case of multiple commits, for convenience. The current behavior is totally fine in that sense.